src/fmtstr.rs

```diff
@@ -115,13 +115,13 @@ impl<'a, 'b> Formatter<'a, 'b> {
         // precision will limit length
         let len = match precision {
             Some(p) => {
-                if p < s.len() {
+                if p < s.chars().count() {
```

Comment:
Store the result of this once at the top, then use it.

This costs `O(n)` every time it is called, which in some cases here can be at least twice.